core/src/com/unciv/ui/components/extensions/Scene2dExtensions.kt

```diff
@@ -130,6 +131,11 @@ fun Actor.getAscendant(predicate: (Actor) -> Boolean): Actor? {
     return null
 }
 
+/** Gets the nearest parent of this actor that is a [T], or null if none of its parents is of that type. */
+inline fun <reified T> Actor.getAscendant(): T? {
```

Comment:
Nice! Actually saw a use of reified in production code last week for exactly this kind of behavior - 'cast as T' - so it's nice to see it again here